src/lib.rs

```diff
@@ -194,13 +194,59 @@ pub enum InterpolationType {
 /// A resampler that us used to resample a chunk of audio to a new sample rate.
 /// The rate can be adjusted as required.
 pub trait Resampler<T>: Send {
-    /// Resample a chunk of audio.
+    /// Resample a chunk of audio. 
+    ///
+    /// The input data is a slice, where each element of the slice is itself referenceable as a slice
+    /// ([`AsRef<[T]>`](AsRef)) which contains the samples for a single channel. Since [`Vec<T>`] implements
+    /// [`AsRef<[T]>`](AsRef), the input may simply be `&*Vec<Vec<T>>`. 
+    /// The output data is a vector, where each element of the vector is itself a vector
+    /// which contains the samples for a single channel.
+    /// TODO! The `active_channels_mask` is optional. (active_channels_mask: Option<&[bool]>)
+    /// Any channel marked as inactive by a false value will be skipped during processing
+    /// and the corresponding output will also be empty.
+    /// If None is given, the length of each input is used to determine the active channels.
+    /// Then if an input channel has zero length, this channel will be considered as inactive. 
```

Comment:
I'm confused about your intention here. Are you intending to add `active_channels_mask` as a parameter to this function or only `process_into_buffer`?

Reply:
Yes I want to add an Option argument for giving a mask, where providing None gives the old behaviour of checking for empty channels.